tests/config/test_config.py

```diff
@@ -42,29 +42,32 @@ def test_init_with_existent_params():
 def test_load_rules_config_file_success(test_files_location):
     mock_rules = ["RuleThatUsesResourceAllowlist", "SecurityGroupOpenToWorldRule"]
     config = Config(stack_name="test_stack", rules=mock_rules)
-    config.load_rules_config_file(open(f"{test_files_location}/config/rules_config_CrossAccountTrustRule.py"))
-    config.add_filters_from_dir(f"{test_files_location}/filters")
-    rule_config = config.get_rule_config("CrossAccountTrustRule")
-    filters = config.get_rule_filters("CrossAccountTrustRule")
-    assert not rule_config.risk_value
-    assert not rule_config.rule_mode
-    assert len(filters) == 1
+    with open(f"{test_files_location}/config/rules_config_CrossAccountTrustRule.py") as f:
```

Comment:
The with should only cover the first line. After that you should release the fd. Same with the rest of the functions (it should also reduce the PR delta quite significantly.

Reply:
Great suggestion! Solved it in: https://github.com/Skyscanner/cfripper/pull/314/commits/287702fcb9d032baef289e46b5855d56e8c72447